azure-quantum/azure/quantum/job/session.py

```diff
@@ -0,0 +1,165 @@
+    from azure.quantum.workspace import Target
+
+class Session(WorkspaceItem):
+    """Azure Quantum Job Session: a logical grouping of jobs.
+
+    :param workspace: Workspace instance to submit job to
+    :type workspace: Workspace
+    :param job_details: Job details model,
+            contains Job ID, name and other details
+    :type job_details: JobDetails
+    """
+
+    def __init__(
+            self,
+            workspace: "Workspace",
+            session_details: Optional[SessionDetails] = None,
+            target: Union[str, "Target", None] = None,
+            provider_id: Optional[str] = None,
+            session_id: Optional[str] = None,
+            session_name: Optional[str] = None,
+            job_failure_policy: Union[str, SessionJobFailurePolicy, None] = None,
+            **kwargs):
+        from azure.quantum.target import Target
+        target_name = target.name if isinstance(target, Target) else target
+        self._target = target if isinstance(target, Target) else None
+
+        if (session_details is not None) and (
+            (isinstance(target, str)) or
+            (provider_id is not None) or
+            (session_id is not None) or
+            (session_name is not None) or
+            (job_failure_policy is not None)):
+            raise ValueError("""If `session_details` is passed, you should not pass `target`, 
+                                `provider_id`, `session_id`, `session_name` or `job_failure_policy`.""")
+
+        if (session_details is None) and (target is None):
+            raise ValueError("If `session_details` is not passed, you should at least pass the `target`.")
+
+        if session_details is None:
+            import uuid
+            import re
+            session_id = session_id if session_id is not None else str(uuid.uuid1())
+            session_name = session_name if session_name is not None else f"session-{session_id}"
+            provider_id = provider_id if provider_id is not None else re.match(r"(\w+)\.", target_name).group(1)
+            session_details = SessionDetails(id=session_id,
+                                             name=session_name,
+                                             provider_id=provider_id,
+                                             target=target_name,
+                                             job_failure_policy=job_failure_policy,
+                                             **kwargs)
+
+        super().__init__(
+            workspace=workspace,
+            details=session_details,
+            **kwargs
+        )
+
+    @property
+    def details(self) -> SessionDetails:
+        return self._details
+
+    @details.setter
+    def details(self, value: SessionDetails):
+        self._details = value
+
+    @property
+    def target(self) -> "Target":
+        return self._target
+
+    def start(self) -> "Session":
+        self.workspace.start_session(self)
+        return self
+
+    def end(self) -> "Session":
```

Comment:
References to "end" and "end_session" need to be updated to "close" and "close_session" everywhere

Reply:
done